From 8b758813d8e2e85ebd2fd4764c02463023ef1204 Mon Sep 17 00:00:00 2001
From: Vladimir Yaichkin <vyaichkin@rzn.dlink.ru>
Date: Thu, 19 Dec 2019 14:39:22 +0300
Subject: twamp, server, service_id: added dlink service connect to twamp
 server
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
Patch-from-hash: b3407d2adbb820bc00cfb54449ac323a307c8bc6
Patch-repo: twamp-protocol
Patch-branch: master
Patch-version: 1

Добавлен вызов функции dlink_service_connect для TWAMP-сервера.

Signed-off-by: Vladimir Yaichkin <vyaichkin@rzn.dlink.ru>
---
 server.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/server.c b/server.c
index a266ac5..e99986f 100644
--- a/server.c
+++ b/server.c
@@ -22,6 +22,10 @@
 #include <time.h>
 #include "twamp.h"

+#ifdef DLINK_SOFTWARE
+#include <d-link/d_service_notify.h>
+#endif
+
 #define MAX_CLIENTS 10
 #define MAX_SESSIONS_PER_CLIENT 10
 #define PORTBASE	20000
@@ -729,6 +733,10 @@ int main(int argc, char *argv[])
     fd_set tmp_fds;
     FD_ZERO(&tmp_fds);

+#ifdef DLINK_SOFTWARE
+    dlink_service_connect(0);
+#endif
+
     int rv;
     while (1) {
         tmp_fds = read_fds;
--
2.20.1

